common/txmgr/types/forwarder_manager.go

```diff
@@ -0,0 +1,12 @@
+package types
+
+import (
+	"github.com/smartcontractkit/chainlink/core/services"
+)
+
+type ForwarderManager[ADDR any] interface {
+	services.ServiceCtx
+	// Name change b/c no distinction b/w EOA and contracts for some chains
+	GetForwarderFor(addr ADDR) (forwarder ADDR, err error)
+	GetForwardedPayload(dest ADDR, origPayload []byte) ([]byte, error)
```

Comment:
Is `GetForwardedPayload` used?

Maybe consider if the `Get` prefixes are actually adding value, although that's pre-existing :shrug: 
```suggestion
	ForwarderAddr(addr ADDR) (forwarder ADDR, err error)
	ForwardedPayload(dest ADDR, origPayload []byte) ([]byte, error)
```

Reply:
Yeah, I believe it does get used in `CreateEthTransaction`. This method is important because it converts the payload to be forwarder-friendly

Reply:
I updated the name to be more understandable.